diff --git a/src/build_context.rs b/src/build_context.rs
index f677bdf..3258a14 100644
--- a/src/build_context.rs
+++ b/src/build_context.rs
@@ -247,13 +247,6 @@ impl BuildContext {
                     }
                 }
                 let new_rpath = new_rpaths.join(":");
-                if let Err(err) = patchelf::set_rpath(&artifact.path, &new_rpath) {
-                    println!(
-                        "⚠️ Warning: Failed to set rpath for {}: {}",
-                        artifact.path.display(),
-                        err
-                    );
-                }
             }
         }
         Ok(())
@@ -308,10 +301,6 @@ impl BuildContext {
             fs::copy(&lib_path, &dest_path)?;
             libs_copied.insert(lib_path);
 
-            patchelf::set_soname(&dest_path, &new_soname)?;
-            if !lib.rpath.is_empty() || !lib.runpath.is_empty() {
-                patchelf::set_rpath(&dest_path, &libs_dir)?;
-            }
             soname_map.insert(
                 lib.name.clone(),
                 (new_soname.clone(), dest_path.clone(), lib.needed.clone()),
@@ -330,9 +319,6 @@ impl BuildContext {
                     }
                 })
                 .collect::<Vec<_>>();
-            if !replacements.is_empty() {
-                patchelf::replace_needed(&artifact.path, &replacements[..])?;
-            }
         }
 
         // we grafted in a bunch of libraries and modified their sonames, but
@@ -346,9 +332,6 @@ impl BuildContext {
                     replacements.push((n, soname_map[n].0.clone()));
                 }
             }
-            if !replacements.is_empty() {
-                patchelf::replace_needed(path, &replacements[..])?;
-            }
             writer.add_file_with_permissions(libs_dir.join(new_soname), path, 0o755)?;
         }
 
@@ -369,7 +352,6 @@ impl BuildContext {
             let new_rpath = Path::new("$ORIGIN").join(relpath(&libs_dir, artifact_dir));
             new_rpaths.push(new_rpath.to_str().unwrap().to_string());
             let new_rpath = new_rpaths.join(":");
-            patchelf::set_rpath(&artifact.path, &new_rpath)?;
         }
         Ok(())
     }
